server/stycoon/game.go

```diff
@@ -2,20 +2,272 @@ package stycoon
+	if err != nil {
+		return ships, fmt.Errorf("query failed %v", err)
+	}
+	var id int
+	var commands map[int]Command
+	if playerId != nil {
+		commands, err = game.getPlayerCommands(*playerId)
+		if err != nil {
+			return ships, err
+		}
+	}
+	for rows.Next() {
+		var ship ShipsValue
+		var pos = make([]int64, 2)
+		var posPrev = make([]int64, 2)
+		err = rows.Scan(&id, &ship.Name, &pos[0], &pos[1], &posPrev[0], &posPrev[1], &ship.ShipClass, &ship.Player, &ship.Life)
+		if err != nil {
+			return ships, fmt.Errorf("row read failed %v", err)
+		}
+		ship.Position = pos
+		ship.PrevPosition = posPrev
+		ship.Resources = game.getResources(id, resources)
+		if playerId != nil {
+			if command, ok := commands[id]; ok {
+				ship.Command = &command
+			}
+		}
+		ships[strconv.Itoa(id)] = ship
+	}
+	if err = rows.Err(); err != nil {
+		return ships, fmt.Errorf("rows read failed: %v", err)
+	}
+	return ships, nil
+}
+
+func (game *Game) getPlayerCommands(playerId int64) (map[int]Command, error) {
+	commands := make(map[int]Command)
+	rows, err := game.db.Query("select object.`id`, `type`, `target`, `resource`, `amount`, `class` from t_command join t_object object on t_command.ship = object.id where object.`owner` = ?", playerId)
+	if err != nil {
+		return commands, err
+	}
+	var id int
+	var target, resource, amount, class sql.NullInt64
+	for rows.Next() {
+		var command Command
+		err = rows.Scan(&id, &command.Type, &target, &resource, &amount, &class)
+		if err != nil {
+			return commands, fmt.Errorf("row read failed %v", err)
+		}
+		if target.Valid {
+			command.Target = target.Int64
+		}
+		if resource.Valid {
+			command.Resource = resource.Int64
+		}
+		if amount.Valid {
+			command.Amount = amount.Int64
+		}
+		if class.Valid {
+			command.ShipClass = class.Int64
+		}
+		commands[id] = command
+	}
+	if err = rows.Err(); err != nil {
+		return commands, fmt.Errorf("rows read failed: %v", err)
+
+	}
+	return commands, nil
+}
+
+func (game *Game) setPlanetResourcePrices(resources *map[int]map[string]*TradingResource) error {
+	rows, err := game.db.Query("select `planet`, `resource`, `buy`, `sell`  from t_price")
+	if err != nil {
+		return fmt.Errorf("query failed %v", err)
+	}
+	var planetId, resourceId int
+	var buy, sell sql.NullFloat64
+	for rows.Next() {
+		err = rows.Scan(&planetId, &resourceId, &buy, &sell)
+		if err != nil {
+			return fmt.Errorf("row read failed %v", err)
+		}
+		resourceIdStr := strconv.Itoa(resourceId)
+		if _, ok := (*resources)[planetId]; !ok {
+			(*resources)[planetId] = make(map[string]*TradingResource)
+		}
+		if _, ok := (*resources)[planetId][resourceIdStr]; !ok {
+			(*resources)[planetId][resourceIdStr] = &TradingResource{}
+		}
+		if buy.Valid {
+			(*resources)[planetId][resourceIdStr].BuyPrice = buy.Float64
+		} else if sell.Valid {
```

Comment:
no `else` here. the planet may be both selling and buying same resource (at different prices)